tools/bin/ci_integration_workflow_launcher.py

```diff
@@ -0,0 +1,185 @@
+    )
+    response.raise_for_status()
+    return run_uuid
+
+
+def get_connector_names():
+    process = subprocess.run(["./gradlew", "integrationTest", "--dry-run"], check=True, capture_output=True, universal_newlines=True)
+
+    res = []
+    for line in process.stdout.splitlines():
+        parts = line.split(":")
+        if (
+            len(parts) >= 4
+            and parts[1] == "airbyte-integrations"
+            and parts[2] in ["connectors", "bases"]
+            and parts[-1] == "integrationTest SKIPPED"
+        ):
+            res.append(parts[3])
+    return res
+
+
+def iter_workflow_runs(owner, repo, per_page=100):
+    path = f"/repos/{owner}/{repo}/actions/runs"
+    page = None
+    while True:
+        params = {"per_page": per_page}
+        if page:
+            params["page"] = page
+        response = get_response(path, params=params)
+        response_json = response.json()
+        for workflow_run in response_json["workflow_runs"]:
+            yield workflow_run
+        if "next" not in response.links:
+            break
+        page = dict(parse_qsl(urlparse(response.links["next"]["url"]).query))["page"]
+
+
+def search_failed_workflow_runs(owner, repo, workflow_id, run_uuids):
+    run_uuids = set(run_uuids)
+    now = datetime.datetime.utcnow()
+    res = set()
+    for workflow_run in iter_workflow_runs(owner, repo):
+        if not run_uuids:
+            break
+
+        created_at = datetime.datetime.strptime(workflow_run["created_at"], "%Y-%m-%dT%H:%M:%SZ")
+        period = now - created_at
+        if period.seconds > 10800:
+            break
+
+        if workflow_run["workflow_id"] != workflow_id:
+            continue
+        if workflow_run["head_branch"] != BRANCH:
+            continue
+        if workflow_run["conclusion"] != "failure":
+            continue
+
+        response_json = get_response_json(workflow_run["jobs_url"])
+        run_uuid = get_run_uuid(response_json["jobs"])
+        if not run_uuid:
+            continue
+
+        if run_uuid in run_uuids:
+            run_uuids.remove(run_uuid)
+            if check_start_aws_runner_failed(response_json["jobs"]):
+                res.add(run_uuid)
+    return res
+
+
+def main():
+    workflow_id = get_workflow_id(ORGANIZATION, REPOSITORY, WORKFLOW_PATH)
+    if not workflow_id:
+        logging.error(f"Cannot find workflow path '{WORKFLOW_PATH}'")
+        sys.exit(1)
+
+    connector_names = get_connector_names()
+    run_uuid_to_name = {}
+    for connector_name in connector_names:
+        run_uuid = workflow_dispatch(ORGANIZATION, REPOSITORY, workflow_id, connector_name)
+        logging.info(f"Dispatch workflow for connector {connector_name}, UUID: {run_uuid}")
+        run_uuid_to_name[run_uuid] = connector_name
+        # to avoid overloading system
+        time.sleep(1)
+
+    logging.info(f"Sleeping {SLEEP} seconds")
+    time.sleep(SLEEP)
```

Comment:
does this mean that if a connector takes longer than 20 minutes to run, it won't be retried? Are we OK with that?

is it possible to have this logic in the integration-test workflow (i.e. maybe it could accept a `auto_retry_count` option or something, and if it fails then it re-triggers itself with one less retry count)? Then we wouldn't need to have this python script doing a long sleep here either.

Reply:
In this script I try to restart ONLY workflows which failed on `AWS Start` step - only this error
If workflows failed on this problem it usually take up to 5 minutes.

Reply:
oops sorry I saw your question only now
yes retries only happens for "AWS Start" problem and only for this problem